x-pack/plugins/ml/public/embeddables/anomaly_charts/initialize_anomaly_charts_controls.ts

```diff
@@ -0,0 +1,112 @@
+ * 2.0.
+ */
+
+import type { TitlesApi } from '@kbn/presentation-publishing/interfaces/titles/titles_api';
+import { BehaviorSubject } from 'rxjs';
+import fastIsEqual from 'fast-deep-equal';
+import type { MlEntityField } from '@kbn/ml-anomaly-utils';
+import type { StateComparators } from '@kbn/presentation-publishing';
+import type { Filter, Query } from '@kbn/es-query';
+import type { JobId } from '../../../common/types/anomaly_detection_jobs';
+import { DEFAULT_MAX_SERIES_TO_PLOT } from '../../application/services/anomaly_explorer_charts_service';
+import type {
+  AnomalyChartsComponentApi,
+  AnomalyChartsDataLoadingApi,
+  AnomalyChartsEmbeddableRuntimeState,
+  AnomalyChartsEmbeddableState,
+} from '../types';
+
+export const initializeAnomalyChartsControls = (
+  rawState: AnomalyChartsEmbeddableState,
+  titlesApi?: TitlesApi,
+  parentApi?: unknown
+) => {
+  const jobIds$ = new BehaviorSubject<JobId[]>(rawState.jobIds);
+  const maxSeriesToPlot$ = new BehaviorSubject<number>(
+    rawState.maxSeriesToPlot ?? DEFAULT_MAX_SERIES_TO_PLOT
+  );
+  const severityThreshold$ = new BehaviorSubject<number | undefined>(rawState.severityThreshold);
+  const selectedEntities$ = new BehaviorSubject<MlEntityField[] | undefined>(
+    rawState.selectedEntities
+  );
+  const interval$ = new BehaviorSubject<number | undefined>(undefined);
+  const dataLoading$ = new BehaviorSubject<boolean | undefined>(true);
+  const blockingError$ = new BehaviorSubject<Error | undefined>(undefined);
+  // Anomaly charts embeddable should consume parent api (i.e. from dashboard global search bar)
+  // since it doesn't need to have its own query and filters
+  // @ts-expect-error Embeddable parentApi has query$
+  const query$ = parentApi?.query$ ?? new BehaviorSubject<Query | undefined>(undefined);
+  // @ts-expect-error Embeddable parentApi has filter$
+  const filters$ = parentApi?.filters$ ?? new BehaviorSubject<Filter[] | undefined>(undefined);
+  // @ts-expect-error Embeddable lastReloadRequestTime has filter$
+  const refresh$ = parentApi?.lastReloadRequestTime$ ?? new BehaviorSubject<void>(undefined);
+
+  const updateUserInput = (update: AnomalyChartsEmbeddableState) => {
+    jobIds$.next(update.jobIds);
+    maxSeriesToPlot$.next(update.maxSeriesToPlot);
+    if (titlesApi) {
+      titlesApi.setPanelTitle(update.title);
+    }
+  };
+
+  const updateSeverityThreshold = (v: number) => severityThreshold$.next(v);
+  const updateSelectedEntities = (v: MlEntityField[]) => selectedEntities$.next(v);
+  const setInterval = (v: number) => interval$.next(v);
+
+  const serializeAnomalyChartsState = (): AnomalyChartsEmbeddableState => {
+    return {
+      jobIds: jobIds$.value,
+      maxSeriesToPlot: maxSeriesToPlot$.value,
+      severityThreshold: severityThreshold$.value,
+      selectedEntities: selectedEntities$.value,
+      filters: filters$.value,
+      query: query$.value,
```

Comment:
if I'm not mistaken query and filer should be handled by the UnifiedSearch api, cc @ThomThomson

Reply:
You're partially correct yes. You shouldn't be serializing filters from the parent into your embeddable _at all_. I'd suggest removing `query` and `filters` from the state of this embeddable, removing all of the behaviour subjects above with the `//@ts-expect-error` comments and using the provided `fetch$` util instead for this.

Reply:
Updated here https://github.com/elastic/kibana/pull/183456/commits/81121a7b00f4688b0a4946c72ce0c6c2a688e0ef